fem/pfespace.cpp

```diff
@@ -1516,18 +1605,36 @@ void ParFiniteElementSpace::GetFaceNbrFaceVDofs(int i, Array<int> &vdofs) const
    }
 }
 
-const FiniteElement *ParFiniteElementSpace::GetFaceNbrFE(int i) const
+const FiniteElement *ParFiniteElementSpace::GetFaceNbrFE(int i, int ndofs) const
 {
-   const FiniteElement *FE =
-      fec->FiniteElementForGeometry(
-         pmesh->face_nbr_elements[i]->GetGeometryType());
-
    if (NURBSext)
    {
       mfem_error("ParFiniteElementSpace::GetFaceNbrFE"
                  " does not support NURBS!");
    }
-   return FE;
+
+   if (ndofs > 0)
+   {
+      for (int order = fec->GetOrder(); ; ++order)
```

Comment:
Even in the current `master` branch, where variable-order spaces are supported in serial, the implementation in `fespace.cpp` heavily uses the assumption that `fec->GetOrder()` is the "base" order. This base order is assumed to be the minimal order and the default, initial order. Generalizing this for p-derefinement below the base order will be a large change that I think should be done in a future PR if needed. This is an issue only if the `FiniteElementCollection` has an initial order larger than the minimum possible order for the space (e.g. 1 in H1). This issue could be avoided by using initial order 1 in H1, and setting all elements to a higher order initially if desired.

Reply:
I see.. Thanks @dylan-copeland 
Do you think then it makes sense to set the `for loop` logic based on whether the initial number of dofs at `fec->GetOrder()` are higher than `ndofs` (check for lower orders down to 0 and break if `ndofs_order < ndofs`) or less than `ndofs` (check for higher orders and break if `ndofs_order > ndofs`, which we currently do)?

Reply:
Line 1632 checks the condition `ndofs_order > ndofs` and aborts in that case, which I think is the appropriate response, since there must be a serious bug or misuse to reach that case (e.g. derefining to an order lower than `fec->GetOrder()`. Note that on the first iteration of the for loop, `ndofs_order` is the number of DOFs for order `fec->GetOrder()`, so the loop will be aborted on the first iteration, if `ndofs` is too small.